Update "Deploying with Now" Guide

timothyis opened the pull request:
<!--
  Q. Which branch should I use for my pull request?
  A. Your best bet is to go for `master`. If you are unsure, ask in the PR, and a Gatsby mantainer will be happy to help :)

  Note: We will only accept bug fixes for Gatsby v1. New features should be added to Gatsby v2.

  Learn more about contributing: https://www.gatsbyjs.org/docs/how-to-contribute/
-->

Hello!

This pull request updates the "Deploying with Now" guide to give information about how to deploy with [Now 2.0](https://zeit.co/blog/now-2). It is a more straight-forward guide than before an holds the most up-to-date information.

Please let me know if I can edit this guide further in any way.
I took example from other guides you have and followed the [styleguide](https://www.gatsbyjs.org/docs/gatsby-style-guide/) as closely as I could. 🙏

jgierer12 commented on docs/docs/deploying-to-now.md:
````diff
@@ -2,36 +2,62 @@
 
-2.  Run `gatsby build` at the root of your project.
+```shell
+npx gatsby new <your project name>
````
AFAIK we deliberately don't use `npx` in the docs since we still support Node 6

jgierer12 commented on docs/docs/deploying-to-now.md:
```diff
@@ -2,36 +2,62 @@
+## Step 2: Getting Now
 
-For large project sizes, it is better to install a Node server package (such as `serve`, or `http-server`):
+If you're not acquainted with Now, it is an easy-to-use cloud platform for serverless deployments that you can use from [the command-line](https://zeit.co/docs/v2/getting-started/installation/#now-cli) or [the desktop app](https://zeit.co/docs/v2/getting-started/installation/#now-desktop)
```
This can be simplified since you already explained what Now is at the beginning of the guide

jgierer12 commented on docs/docs/deploying-to-now.md:
```diff
@@ -2,36 +2,62 @@
-1.  Install a Node server package
+We recommend that you download and [install Now Desktop](https://zeit.co/download) which installs Now CLI and keeps it up-to-date automatically.
+
+To install Now CLI quickly, use the following:
```
For brevity, I think we should focus on one of the installation methods here and just quickly mention the other, e.g.

```markdown
We recommend that you download and [install Now Desktop](https://zeit.co/download) which installs Now CLI and keeps it up-to-date automatically. You can also [install Now CLI via npm](https://example.com).
```

jgierer12 commented on docs/docs/deploying-to-now.md:
```diff
@@ -2,36 +2,62 @@
   "scripts": {
-    "start": "serve public/"
+    ...
+    "now-build": "npm run build"
```
```suggestion
    "now-build": "gatsby build"
```

Better make this package manager independent, since some may prefer using Yarn.

DSchau replied:
@jgierer12 I don't actually agree with this change! We want to run the `build` script here in case there is anything (e.g. cleanup, setup, etc.) that is built into the build script.

I generally use `npm run` even though I am yarn user, and I think this makes sense since _every_ yarn user will also have npm, and I can't see a benefit to using `yarn run [task]`

jgierer12 replied:
That's true, I sometimes see people completely avoiding `npm` but in this case it actually makes sense.

jgierer12 commented on docs/docs/deploying-to-now.md:
```diff
@@ -2,36 +2,62 @@
+## Step 4: Deploying
+
+You can deploy your application by running the following in the root of the project directory, where the `now.json` is:
+`now`
```
~~~markdown
```bash
now
```
~~~

timothyis commented:
Thanks for the review @jgierer12! I've addressed your comments, however, I opted to keep a highlighted spot for installing Now CLI via npm since it's often a choice a user wants to take to get through a tutorial as quickly as possible, I've found.

I'm happy to change that again if you'd prefer though!

DSchau commented on docs/docs/deploying-to-now.md:
```diff
@@ -2,36 +2,67 @@
 ---
 
-In order to deploy your Gatsby project using [Now](https://zeit.co/now), you can do the following:
+[ZEIT Now](https://zeit.co/now) is a [cloud platform for serverless deployment](https://zeit.co/docs/v2/getting-started/introduction-to-now/) that you can use to deploy your Gatsby projects and [alias them](https://zeit.co/docs/v2/domains-and-aliases/aliasing-a-deployment/) to your domain or a free `.now.sh` suffixed URL.
```
```suggestion
[ZEIT Now](https://zeit.co/now) is a [cloud platform for serverless deployment](https://zeit.co/docs/v2/getting-started/introduction-to-now/) that you can use to deploy your Gatsby projects and [alias them](https://zeit.co/docs/v2/domains-and-aliases/aliasing-a-deployment/) to your personal domain or a free `.now.sh` suffixed URL.
```

DSchau commented on docs/docs/deploying-to-now.md:
````diff
@@ -2,36 +2,67 @@
+```
 
-2.  Add a `start` script to your `package.json` file. This is what Now will use to run your application:
+This will allow Now to understand that we want to use the [latest Now 2.0 version](https://zeit.co/blog/now-2) of [the platform](https://zeit.co/docs/v2/platform/overview/), to set the project name to be `my-gatsby-project`, and to use the [@now/static-build builder](https://zeit.co/docs/v2/deployments/official-builders/static-build-now-static-build/) to take the `package.json` file as an entrypo ...
````
Can we re-visit this? It's quite a sentence. I think it may be helpful to break this out into a list, e.g.

```suggestion
This `now.json` file will allow us to do several things, specifically:

- Use the [latest Now 2.0 version](https://zeit.co/blog/now-2) of [the platform](https://zeit.co/docs/v2/platform/overview/)
- Set the project name to `my-gatsby-project`
- Use the [@now/static-build builder](https://zeit.co/docs/v2/deployments/official-builders/static-build-now-static-build/) to take the `package.json` file as an entrypoint and use the `public` directory as the our content directory
```

DSchau commented on docs/docs/deploying-to-now.md:
```diff
@@ -2,36 +2,67 @@
-2.  Add a `start` script to your `package.json` file. This is what Now will use to run your application:
+This will allow Now to understand that we want to use the [latest Now 2.0 version](https://zeit.co/blog/now-2) of [the platform](https://zeit.co/docs/v2/platform/overview/), to set the project name to be `my-gatsby-project`, and to use the [@now/static-build builder](https://zeit.co/docs/v2/deployments/official-builders/static-build-now-static-build/) to take the `package.json` file as an entrypo ...
 
+The final setup step is to add a script to the `package.json` which will build our application:
```
```suggestion
The final step is to add a script to the `package.json` which will build our application:
```

DSchau commented on docs/docs/deploying-to-now.md:
```diff
@@ -2,36 +2,67 @@
+
+## References:
+- [Deploying Gatsby with Now](https://zeit.co/examples/gatsby/)
+
```
```suggestion
```

DSchau replied:
(removing an extra line)

timothyis commented:
@DSchau I did include a reference to it at the bottom, as I saw your s3 deployment guide had something similar, however, I'm happy to incorporate it more into the content if you'd like.
https://github.com/gatsbyjs/gatsby/pull/10390/files#diff-847fa9e659a8af2357ccc57be66bc3cdR70

DSchau commented:
@timothyis I missed it! That's perfect, makes a lot of sense.

Merging; thanks so much for the PR 🎉

timothyis commented:
Thank you!

shannonbux commented on docs/docs/deploying-to-now.md:
````diff
@@ -2,36 +2,79 @@
+
+With Now CLI installed, we can go on to deploy our previously setup Gatsby project by first creating a `now.json` file with the following contents:
+
+```json:title=now.json
````
is this at the root of the project?

shannonbux commented on docs/docs/deploying-to-now.md:
```diff
@@ -2,36 +2,79 @@
+- Set the project name to `my-gatsby-project`
+- Use the [@now/static-build builder](https://zeit.co/docs/v2/deployments/official-builders/static-build-now-static-build/) to take the `package.json` file as an entrypoint and use the `public` directory as the our content directory
+
+The final step is to add a script to the `package.json` which will build our application:
```
*your application
